src/ripple/rpc/impl/ServerHandlerImp.cpp

```diff
@@ -540,185 +548,194 @@ ServerHandlerImp::processRequest (Port const& port,
 
-    usage.charge (loadType);
-    if (usage.warn())
-        result[jss::warning] = jss::load;
-
-    Json::Value reply (Json::objectValue);
-    reply[jss::result] = std::move (result);
-    if (jsonRPC.isMember(jss::jsonrpc))
-        reply[jss::jsonrpc] = jsonRPC[jss::jsonrpc];
-    if (jsonRPC.isMember(jss::ripplerpc))
-        reply[jss::ripplerpc] = jsonRPC[jss::ripplerpc];
-    if (jsonRPC.isMember(jss::id))
-        reply[jss::id] = jsonRPC[jss::id];
-    auto response = to_string (reply);
-
-    rpc_time_.notify (static_cast <beast::insight::Event::value_type> (
-        std::chrono::duration_cast <std::chrono::milliseconds> (
-            std::chrono::high_resolution_clock::now () - start)));
-    ++rpc_requests_;
-    rpc_size_.notify (static_cast <beast::insight::Event::value_type> (
-        response.size ()));
-
-    response += '\n';
-
-    if (auto stream = m_journal.debug())
-    {
-        static const int maxSize = 10000;
-        if (response.size() <= maxSize)
-            stream << "Reply: " << response;
+        // Always report "status".  On an error report the request as received.
+        bool on_error = false;
+        if (result.isMember (jss::error))
+        {
+            result[jss::status] = jss::error;
+            result[jss::request] = params;
+            JLOG (m_journal.debug())  <<
+                "rpcError: " << result [jss::error] <<
+                ": " << result [jss::error_message];
+            on_error = true;
+        }
         else
-            stream << "Reply: " << response.substr (0, maxSize);
-    }
+        {
+            result[jss::status]  = jss::success;
+        }
 
-    HTTPReply (200, response, output, rpcJ);
+        usage.charge (loadType);
+        if (usage.warn())
+            result[jss::warning] = jss::load;
+
+        Json::Value reply (Json::objectValue);
+        reply[jss::result] = std::move (result);
+        if (jsonRPC.isMember(jss::jsonrpc))
+            reply[jss::jsonrpc] = jsonRPC[jss::jsonrpc];
+        if (jsonRPC.isMember(jss::ripplerpc))
+            reply[jss::ripplerpc] = jsonRPC[jss::ripplerpc];
+        if (jsonRPC.isMember(jss::id))
+            reply[jss::id] = jsonRPC[jss::id];
+        auto response = to_string (reply);
+
+        rpc_time_.notify (static_cast <beast::insight::Event::value_type> (
+            std::chrono::duration_cast <std::chrono::milliseconds> (
+                std::chrono::high_resolution_clock::now () - start)));
+        ++rpc_requests_;
+        rpc_size_.notify (static_cast <beast::insight::Event::value_type> (
+            response.size ()));
+
+        response += '\n';
+
+        if (auto stream = m_journal.debug())
+        {
+            static const int maxSize = 10000;
+            if (response.size() <= maxSize)
+                stream << "Reply: " << response;
+            else
+                stream << "Reply: " << response.substr (0, maxSize);
+        }
+
+        HTTPReply (200, response, output, rpcJ);
```

Comment:
For every HTTP request, there must be exactly one non-Informational response. Informational responses are indicated with a status code from 100 to 199 inclusive.